### PR TITLE
feat: Allow disabling securityContext

### PR DESCRIPTION
**Pull request by fraxachun**
Allows setting `securityContext.enabled: false` which omits the securityContext-setting for the containers. In some environment (e.g. OpenShift) you don't want to set them (e.g. they are set automatically).

## Related Issue or Design Document

https://github.com/ory/k8s/issues/421

## Checklist

- [X] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [X] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [X] I have read the [security policy](../security/policy).
- [X] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [X] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Comment by Demonsthere** [comment]
Hi there!
I think this is a bit of a overkill 😅  in your values files you can just specify `securityContext: {}` to be an empty object, which should override the default values

**Comment by fraxachun** [comment]
In my experience (helm v3.8.1) settting `securityContext: {}` does not work because helm is merging it with the default values. `securityContext: null` also does not work (https://github.com/helm/helm/issues/9136). I also tried `securityContext: false`, but this leads to complaining helm about the datatype and the value is rendered.

**Comment by Demonsthere** [comment]
I see, and I assume you're using the charts in an umbrella chart which has the rest as sub-charts, and setting `null` does not propagate for you?

We can make `securityContext: false` work, but need to modify your PR and include `if` clauses like
```yaml
          {{- if .Values.securityContext }}
          securityContext:
            {{- toYaml .Values.securityContext | nindent 12 }}
          {{- end }}
```

File: helm/charts/hydra-maester/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ A Helm chart for Kubernetes
 | deployment.nodeSelector | object | `{}` | Node labels for pod assignment. |
 | deployment.podLabels | object | `{}` |  |
 | deployment.resources | object | `{}` |  |
-| deployment.securityContext | object | `{"allowPrivilegeEscalation":false,"capabilities":{"drop":["ALL"]},"privileged":false,"readOnlyRootFilesystem":true,"runAsNonRoot":true,"runAsUser":1000}` | Default security context |
+| deployment.securityContext | object | `{"enabled": true",allowPrivilegeEscalation":false,"capabilities":{"drop":["ALL"]},"privileged":false,"readOnlyRootFilesystem":true,"runAsNonRoot":true,"runAsUser":1000}` | Default security context |
```
Comment (Demonsthere) [inline helm/charts/hydra-maester/README.md:22]: Those files are auto-generated, please do not modify them by hand :)

File: helm/charts/hydra-maester/templates/deployment.yaml [hunk 2]
```diff
@@ -54,8 +54,10 @@ spec:
             {{- toYaml .Values.deployment.resources | nindent 12 }}
           terminationMessagePath: /dev/termination-log
           terminationMessagePolicy: File
+          {{- if .Values.deployment.securityContext.enabled }}
           securityContext:
-            {{- toYaml .Values.deployment.securityContext | nindent 12 }}
+            {{- omit .Values.deployment.securityContext "enabled" | toYaml | nindent 12 }}
```
Comment (Demonsthere) [inline helm/charts/hydra-maester/templates/deployment.yaml:59]:
```suggestion
            {{- toYaml .Values.deployment.securityContext | nindent 12 }}
```

Comment (Demonsthere) [reply helm/charts/hydra-maester/templates/deployment.yaml:59]: Please see the PR discussion

**Comment by fraxachun** [comment]
Technically setting `securityContext: false` would work but helm complains about wrong datatypes:
```
~/dev/k8s master$ helm template test helm/charts/hydra --set deployment.securityContext=false
coalesce.go:220: warning: cannot overwrite table with non table for hydra.deployment.securityContext (map[allowPrivilegeEscalation:false capabilities:map[drop:[ALL]] privileged:false readOnlyRootFilesystem:true runAsNonRoot:true runAsUser:100])
coalesce.go:220: warning: cannot overwrite table with non table for hydra.deployment.securityContext (map[allowPrivilegeEscalation:false capabilities:map[drop:[ALL]] privileged:false readOnlyRootFilesystem:true runAsNonRoot:true runAsUser:100])
```

I would prefer the suggested solution. However, if this behaviour is acceptable I can update my PR with the `securityContext: false` solution.

**Comment by Demonsthere** [comment]
Yes, please update the PR :) I think that the `securityContext: false` is a good middle point solution.

**Comment by fraxachun** [comment]
@Demonsthere I force-pushed the suggested changes.

**Review by Demonsthere — APPROVED** [review]
Great, thanks! :)

**Pull request merged**